### ISSUE TITLE
plan with 0.10.4 doesn't show (forces new resource) when a resource is recreated

### ISSUE THREAD

**Issue by defn** [hashicorp/terraform#16035]
Terraform 0.10.4's plan no longer displays the `(forces new resource)` next to the changed attribute when a resource is recreated.   Makes it hard to scan for why a resource changed. 

### Terraform Version
```
> terraform -v
Terraform v0.10.4
```

### Terraform Configuration Files
```hcl
resource "local_file" "example" {
  content  = "changeme"
  filename = "example.txt"
}
```
### Expected Behavior
When I change the content of the local_file resource, `terraform plan` should show that a new resource is required, and which attribute forced the change.

With 0.10.3 and a change to the local_file content:

```
-/+ local_file.example (new resource required)
      content:  "changeme" => "changeme!!!!!!" (forces new resource)
      filename: "example.txt" => "example.txt"
Plan: 1 to add, 0 to change, 1 to destroy.
```

### Actual Behavior
With terraform 0.10.4, the plan output is:

```
-/+ local_file.example (new resource required)
      id:       "fa9beb99e4029ad5a6615399e7bbae21356086b3" => <computed>
      content:  "changeme" => "changeme!!!!!!"
      filename: "example.txt" => "example.txt"
```

It doesn't have the `"(forces new resource)"` at the end of the line that contains `content:`

### Steps to Reproduce
Using terraform 0.10.4:

1. `terraform init`
2. `terraform apply`
3. Change contents of local_file resource in main.tf
4. `terraform plan`

**Comment by farvour** [comment]
Confirmed. Used to see this in terraform 0.10.2, upgraded to 0.10.4 today and no longer is the "why" a new resource being force shown next to the relevant attribute.

**Comment by apparentlymart** [comment]
Thanks for reporting this @defn, and sorry for this regression.

This was a mistake during some refactoring of the plan renderer in 0.10.4, not quite adapting the existing code to the new interface correctly. I've pushed a fix in #16067 so hopefully we can get this fixed in the next release.
